clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -670,10 +662,11 @@ private void validateCooperativeAssignment(final Map<String, List<TopicPartition
     }
 
     @Override
-    protected void onJoinPrepare(int generation, String memberId) {
+    protected void onJoinPrepare(int generation, String memberId, final Timer pollTimer) {
         log.debug("Executing onJoinPrepare with generation {} and memberId {}", generation, memberId);
         // commit offsets prior to rebalance if auto-commit enabled
-        maybeAutoCommitOffsetsSync(time.timer(rebalanceConfig.rebalanceTimeoutMs));
+        //The timer whose commitOffset timed out is no longer time.timer(rebalanceConfig.rebalanceTimeoutMs), and is changed to the timer passed by the customer
+        maybeAutoCommitOffsetsSync(pollTimer);
```

Comment:
Did we have any null handling for the `pollTimer`? Maybe we can pass the old `time.timer(rebalanceConfig.rebalanceTimeoutMs)` for null case here?

Reply:
Thank you for your reminder. I suggest that when pollTimer is null, it is safer to pass `time.timer(0L)` to `maybeAutoCommitOffsetsSync` instead of passing the old `time.timer(rebalanceConfig.rebalanceTimeoutMs)`

`time.timer(0L)` indicates that if allConsumedOffsets is not empty, submit the offset synchronously at least once

Reply:
Because even if we have the `cleanUpConsumedOffsets` method, we may encounter a non-`UnknownTopicOrPartitionException` **retryable** type exception  in the `commitOffsetsSync(Map<TopicPartition, OffsetAndMetadata> offsets, Timer timer)` method. If we pass in time.timer(rebalanceConfig. rebalanceTimeoutMs), it may trigger the case in jira again.